File: pms-backend/src/main/java/com/pms/service/PatientService.java
package com.pms.service;

import com.pms.dto.PatientDto;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface PatientService {
    PatientDto createPatient(PatientDto dto);
    PatientDto updatePatient(Long id, PatientDto dto);
    PatientDto getPatientById(Long id);
    PatientDto getPatientByUserId(Long userId);
    List<PatientDto> getAllPatients();
    Page<PatientDto> searchPatients(String query, String status, String gender, Pageable pageable);
    void deletePatient(Long id);
}

File: pms-frontend/src/services/patientService.js
import api from './api';

export const patientService = {
  getAllPatients: async () => {
    const res = await api.get('/patients');
    return res.data;
  },

  searchPatients: async (params = {}) => {
    const res = await api.get('/patients/search', { params });
    return res.data;
  },

  getPatientById: async (id) => {
    const res = await api.get(`/patients/${id}`);
    return res.data;
  },

  getPatientByUserId: async (userId) => {
    const res = await api.get(`/patients/user/${userId}`);
    return res.data;
  },

  createPatient: async (data) => {
    const res = await api.post('/patients', data);
    return res.data;
  },

  updatePatient: async (id, data) => {
    const res = await api.put(`/patients/${id}`, data);
    return res.data;
  },

  deletePatient: async (id) => {
    const res = await api.delete(`/patients/${id}`);
    return res.data;
  }
};
